google/cloud/spark_connect/session.py

```diff
@@ -73,6 +74,7 @@ class GoogleSparkSession(SparkSession):
     _region = None
     _client_options = None
     _active_s8s_session_id: ClassVar[Optional[str]] = None
+    _installed_artifacts: set[str] = set()
```

Comment:
Let's not do de-duping as this may have side-effects that hard to control for. We can add it later on if we really need it.

Reply:
Discussing in other thread.